src/Data/Scope.php

```diff
@@ -0,0 +1,67 @@
+<?php
+namespace LORIS\Data;
+
+/**
+ * A Scope is an enumeration class which represents the scope
+ * that a piece of data may apply to in LORIS.
+ *
+ * The Scope class is final because the list of enumeration types
+ * can not be dynamically extended without modifying all places
+ * that must deal with the enumeration options.
```

Comment:
this seems limiting to projects wanting to define their own scopes... why not a database table defining the constants...

would avoid overriding/overloading specially since we dont offer any easy overriding structure for the `src` directory

Reply:
Scopes can't be extensible because everywhere in the code that deals with the data needs to understand how to deal with that scope. Let's say a module added a scope named "foobar" and someone accessed that a variable with that scope in the DQT. Questions like "how should this be displayed? How does it relate back to candidates? How does it relate to sessions? Does it need to show a list of sessions to select? How does it get displayed in the results table? How do I join it with candidate scoped data to show in a row?" need to be understood by the code to build the interface and display results.

Reply:
(if you have a suggestion for how to make the comment more clear I'll update it.)

Reply:
no your right... I'm just thinking if there is a way we can add project defined scopes in the same way. my immediate interest is the Biospecimen scope for CBIGR

Reply:
it could come later though

Reply:
There are similar problems with things like imaging QC. "Pass/Fail" is really file scoped data but there is no file scope. For now I've defined them as "many" cardinality scoped to the session.